docs/metric_deletion_retention.md

```diff
@@ -0,0 +1,47 @@
+# Metric Deletion 
```

Comment:
I'd add a paragraph above as an intro outlining the different types of deletions we support. Namely deletion of entire series or deletion of old data in individual metrics.